scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowDoFns.scala

```diff
@@ -75,38 +80,116 @@ private[this] abstract class PredictDoFn[T, V, M <: Model[_]](
 
     c.output(result)
   }
-
-  override def createResource(): ConcurrentMap[String, M] =
-    new ConcurrentHashMap[String, M]()
 }
 
-private[tensorflow] class SavedBundlePredictDoFn[T, V](
+private[tensorflow] abstract class SavedBundlePredictDoFn[T, V](
   uri: String,
-  options: TensorFlowModel.Options,
-  fetchOp: Seq[String],
-  inFn: T => Map[String, Tensor[_]],
-  outFn: (T, Map[String, Tensor[_]]) => V
-) extends PredictDoFn[T, V, TensorFlowModel](fetchOp, inFn, outFn) {
+  signatureName: String,
+  options: TensorFlowModel.Options
+) extends PredictDoFn[T, V, TensorFlowModel] {
   @transient private lazy val log = LoggerFactory.getLogger(this.getClass)
 
-  override def withRunner(f: Session#Runner => V): V = {
-    val model = getResource
-      .computeIfAbsent(
-        uri,
-        new Function[String, TensorFlowModel] {
-          override def apply(t: String): TensorFlowModel =
-            Models
-              .tensorFlow(uri, options)
-              .get(Duration.ofDays(Integer.MAX_VALUE))
-        }
-      )
+  override def withRunner(f: Session#Runner => V): V = f(getResource.instance().session().runner())
 
-    f(model.instance().session().runner())
-  }
+  def id(): String =
+    (Seq("tf", uri, signatureName) ++ options.tags.asScala.mkString(";")).mkString(":")
```

Comment:
This isn't particularly useful, since the name is only internal to the loaded model